src/utils/ucx/ucx_utils.cpp

```diff
@@ -318,32 +318,22 @@ nixlUcxContext::nixlUcxContext(std::vector<std::string> devs,
                                size_t req_size,
                                nixlUcxContext::req_cb_t init_cb,
                                nixlUcxContext::req_cb_t fini_cb,
-                               nixl_ucx_mt_t __mt_type,
-                               bool prog_thread)
+                               bool prog_thread,
+                               unsigned long num_workers,
+                               nixl_thread_sync_t sync_mode)
 {
     ucp_params_t ucp_params;
     ucp_config_t *ucp_config;
     ucs_status_t status = UCS_OK;
 
-    mt_type = __mt_type;
+    mt_type = (sync_mode >= nixl_thread_sync_t::NIXL_THREAD_SYNC_RW || prog_thread) ?
```

Comment:
This doesn't seem a right comparison given the threading modes declaration:
```C
enum class nixl_thread_sync_t {
    NIXL_THREAD_SYNC_NONE,
    NIXL_THREAD_SYNC_STRICT,
    NIXL_THREAD_SYNC_RW,
    NIXL_THREAD_SYNC_DEFAULT = NIXL_THREAD_SYNC_NONE,
};
```

Should it be 
```
sync_mode > nixl_thread_sync_t::NIXL_THREAD_SYNC_NONE
```
?

Reply:
I'd rather vote for some kind of more reliable statement, like maybe having a `NIXL_THREAD_NEED_LOCKS()` macro defined next to `nixl_thread_sync_t` and just used here to avoid similar mistakes in future (if this is indeed a mistake, of course)

Reply:
What is the mistake here? In the following code the result of this conditional is mapped to one of the following:
```
    UCS_THREAD_MODE_SERIALIZED, /**< Multiple threads can access, but only one at a time */
    UCS_THREAD_MODE_MULTI,      /**< Multiple threads can access concurrently */
```

Unless I'm missing something, simultaneous access from multiple threads to a worker instance is only possible either when the locking model is permissive (e.g. multiple concurrent user data path threads) or when the progress thread is enabled (which will run concurrently with user data path threads irrespective of the synchronization model chosen).

Reply:
do I understand correctly that with `NIXL_THREAD_SYNC_STRICT` thread safety will be ensured on NIXL level using NIXL_LOCK_GUARD? And we do not need to have mt support in UCX then?
in general I agree with @artpol84 that current initialization of `mt_type` is confusing, can you pls add some comment or initialize it somehow more explicitly?